### ISSUE TITLE
setInterval, setTimeout functions

### ISSUE THREAD

**Issue by rsredsq** [AtomicGameEngine/AtomicGameEngine#162]
Just need a realisation of `setInterval`, `setTimeout` functions. Take look at here: https://github.com/svaarala/duktape/tree/master/examples/eventloop

**Comment by shaddockh** [comment]
To add to this, setImmediate should also be implemented.  These basically give you the ability to defer an action via a callback to the end of the current thread of execution. (ie, at the end of an update cycle)

```int setInterval( callback, interval)```  - Allows you to schedule a block of code to execute every x milliseconds and repeat
```int setTimeout( callback, milliseconds)```  - Allows you to schedule a one-off block of code to execute in x milliseconds from now
```int setImmediate( callback )``` - Allows you to schedule a one-off block of code to execute as soon as the current thread of execution is complete
```clearInterval(id)``` - cancels a setInterval with the provided id
```clearTimeout(id)``` - cancels a setTimeout with the provided id
```clearImmediate(id)``` - cancels a setImmediate with the provided it

For an example of how this is useful, say you have a component that broadcasts a message to various other components.  One of those components will delete a resource that the other components may need within their message handler.  Ideally, what you would want to do is defer the deletion of that resource to the end of the event loop.

pseudocode
```
trigger(node, 'onHit');

COMPONENT1:
function onHit() {
  // defer deltion until the end of the current update cycle
  var nodeToDelete = this.node;
   setImmediate(function() { Atomic.destroy(nodeToDelete); });
}

COMPONENT2:
function onHit() {
  // display something about the node to the screen
}
```

Note, that having these functions also open up the ability to pull in 3rd party libraries that depend on these functions such as Promises, etc.

**Comment by JoshEngebretson** [comment]
Nice detailed description!  Bumping in priority

**Comment by JoshEngebretson** [comment]
Closed via #259 with example: https://github.com/AtomicGameEngine/AtomicExamples/tree/master/EventLoop
